pkg/cmd/gitprovider/add.go

```diff
@@ -44,9 +49,84 @@ var GitProviderAddCmd = &cobra.Command{
 		setGitProviderConfig.Username = new(string)
 		setGitProviderConfig.Alias = new(string)
 
-		err = gitprovider_view.GitProviderCreationView(ctx, apiClient, &setGitProviderConfig, existingAliases)
-		if err != nil {
-			return err
+		if len(args) == 0 {
+			err = gitprovider_view.GitProviderCreationView(ctx, apiClient, &setGitProviderConfig, existingAliases)
+			if err != nil {
+				return err
+			}
+		} else {
+			supportedProviders := config.GetSupportedGitProviders()
+			supportedProviderIds := util.ArrayMap(supportedProviders, func(gp config.GitProvider) string {
+				return gp.Id
+			})
+
+			if slices.Contains(supportedProviderIds, args[0]) {
+				setGitProviderConfig.ProviderId = args[0]
+			}
+			providerId := setGitProviderConfig.ProviderId
+
+			if providerId == "" {
+				supportedProvidersStr := strings.Join(supportedProviderIds, ", ")
+				return fmt.Errorf("'%s' is invalid or not a supported git provider.\nSupported providers are: %s", args[0], supportedProvidersStr)
+			}
+
+			if tokenFlag == "" {
+				return fmt.Errorf("token is required")
+			}
+			setGitProviderConfig.Token = tokenFlag
+
+			if aliasFlag != "" {
+				for _, alias := range existingAliases {
+					if alias == aliasFlag {
+						initialAlias := setGitProviderConfig.Alias
+						if initialAlias == nil || *initialAlias != aliasFlag {
+							return fmt.Errorf("alias '%s' is already in use", aliasFlag)
+						}
+					}
+				}
+				setGitProviderConfig.Alias = &aliasFlag
+			}
+
+			if gitprovider_view.ProviderRequiresUsername(providerId) {
```

Comment:
I suggest that we throw an error if the user sets properties that the GP does not support.

e.g. setting base api url for github should throw an error.